src/findCommand.js

```diff
@@ -0,0 +1,152 @@
+import Command from '@ckeditor/ckeditor5-core/src/command';
+import { CURRENT_SEARCH_MARKER, isSameSearch, removeCurrentSearchMarker, removeSearchMarkers, SEARCH_MARKER } from './utils';
+
+const DEFAULT_OPTIONS = {
+	findText: '',
+	increment: 1,
+	replaceText: '',
+	replaceAll: false
+};
+
+export default class FindCommand extends Command {
+    constructor(editor) {
+        super(editor);
+        this.currentSearchIndex = 0;
+    }
+    execute( userOptions ) {
+		if ( !userOptions.findText ) {
+			return;
+		}
+		const options = { ...DEFAULT_OPTIONS, ...userOptions };
+		if ( options.replaceText ) {
+			if ( options.replaceAll ) {
+				return this._replaceAll( options.findText, options.replaceText );
+			}
+			return this._replace( options.findText, options.replaceText );
+		}
+		return this._find( options.findText, options.increment );
+    }
+
+    _find(searchText, increment){
+    	const editor = this.editor;
+    	const model = editor.model;
+    	let markers = Array.from( model.markers.getMarkersGroup( SEARCH_MARKER ) );
+
+		if ( isSameSearch( searchText, markers ) ) {
+			// loop through the items
+			this.currentSearchIndex = ( this.currentSearchIndex + markers.length + increment ) % markers.length;
+		}
+		else {
+			this._resetStatus();
+			// Create a range spanning over the entire root content:
+            const range = model.createRangeIn( model.document.getRoot() );
+			let counter = 0;
+			model.change( writer => {
+				// Iterate over all items in this range:
+				for ( const value of range.getWalker() ) {
+					const textNode = value.item.textNode;
+					if ( textNode ) {
+						const text = value.item.data;
+						const indices = getIndicesOf( searchText, text, false );
+						for ( const index of indices ) {
+							const label = SEARCH_MARKER + ':' + searchText + ':' + counter++;
+							const startIndex = textNode.startOffset + index;
+							const start = writer.createPositionAt( textNode.parent, startIndex );
+							const end = writer.createPositionAt( textNode.parent, startIndex + searchText.length );
+							const range = writer.createRange( start, end );
+							writer.addMarker( label, { range, usingOperation: false } );
+						}
+					}
+				}
+				// update markers variable after search
+				markers = Array.from( model.markers.getMarkersGroup( SEARCH_MARKER ) );
+			} );
+			this.currentSearchIndex = 0;
+		}
+
+		const currentMarker = markers[ this.currentSearchIndex ];
+		this._scrollTo( currentMarker );
+		return {
+			currentMarker,
+			markers,
+			currentIndex:this.currentSearchIndex,
+			total: markers.length,
+		};
+    }
+
+    _replace(findText,replaceText ) {
+		const model = this.editor.model;
+		const markers = Array.from( model.markers.getMarkersGroup( SEARCH_MARKER ) );
+		const sameSearch = isSameSearch( findText, markers );
+		const currentMarker = sameSearch ? markers[ this.currentSearchIndex ] : this._find( findText, 1 );
```

Comment:
maybe the problem with replace not being fired is here, now that `_find` returns an object, not only the currentMarker. I think destructure may fix this.